swiftshield-Sources/Protector.swift

```diff
@@ -70,9 +70,15 @@ class Protector {
             let actionModule = idToXML[element.attributes["destination"] ?? ""]?.attributes["customModule"] ?? ""
             if supportedModules?.contains(actionModule) != false {
                 Logger.log(.protectedReference(originalName: trueName, protectedName: protectedClass))
-                element.attributes["selector"] = protectedClass + ":"
+                let tmp = actionSelector.contains(":") ? ":" : ""
+                element.attributes["selector"] = protectedClass + tmp
+            } else if element.parent?.name == "connections", element.parent?.parent?.name.contains("GestureRecognizer") != false {
+                Logger.log(.protectedReference(originalName: trueName, protectedName: protectedClass))
+                let tmp = actionSelector.contains(":") ? ":" : ""
+                element.attributes["selector"] = protectedClass + tmp
```

Comment:
Because they both do the same thing, we can put everything together:
```
let isSelectorReference = supportedModules?.contains(actionModule) != false
let isGestureReference = element.parent?.name == "connections", element.parent?.parent?.name.contains("GestureRecognizer") != false
if isSelectorReference || isGestureReference {
    Logger.log(.protectedReference(originalName: trueName, protectedName: protectedClass))
    let tmp = actionSelector.contains(":") ? ":" : ""
    element.attributes["selector"] = protectedClass + tm
}
```